documentation/docs/02-runes/02-$state.md

```diff
@@ -111,6 +137,165 @@ person = {
+
+```js
+// @errors: 7006
+function createTodo(initial) {
+	let text = initial;
+	let done = false;
+	return {
+		text,
+		done,
+		log: () => console.log(text, done)
+	}
+}
+
+const todo = createTodo('wrong');
+todo.log(); // logs "'wrong', false"
+todo.done = true;
+todo.log(); // still logs "'wrong', false"
+```
+
+The value change does not propagate back into the function body of `createTodo`, because `text` and `done` are read at the point of return, and are therefore a fixed value. To make that work, we have to bring the read and write into the scope of the function body. This can be done via getter/setters or via function calls:
+
+```js
+// @errors: 7006
+function createTodo(initial) {
+	let text = initial;
+	let done = false;
+	return {
+		// using getter/setter
+		get text() { return text },
+		set text(v) { text = v },
+		// using functions
+		isDone() { return done },
+		toggle() { done = !done },
+		// log
+		log: () => console.log(text, done)
+	}
+}
+
+const todo = createTodo('right');
+todo.log(); // logs "'right', false"
+todo.text = 'changed'; // invokes the setter
+todo.toggle(); // invokes the function
+todo.log(); // logs "'changed', true"
+```
+
+What you could also do is to instead create an object and return that as a whole. While the variable itself is fixed in time, its properties are not, and so they can be changed from the outside and the changes are observable from within the function:
+
+```js
+// @errors: 7006
+function createTodo(initial) {
+	const todo = { text: initial, done: false }
+	return {
+		todo,
+		log: () => console.log(todo.text, todo.done)
+	}
+}
+
+const todo = createTodo('right');
+todo.log(); // logs "'right', false"
+todo.todo.done = true; // mutates the object
+todo.log(); // logs "'right', true"
+```
+
+Classes are similar, their properties are "live" due to the `this` context:
+
+```js
+// @errors: 7006
+class Todo {
+	done = false;
+	text;
+
+	constructor(text) {
+		this.text = text;
+	}
+
+	log() {
+		console.log(this.done, this.text)
+	}
+}
+
+const todo = new Todo('right');
+todo.log(); // logs "'right', false"
+todo.done = true;
+todo.log(); // logs "'right', true"
+```
+
+Notice how we didn't use _any_ Svelte specifics, this is just regular JavaScript semantics. `$state` and `$state.raw` (and [`$derived`]($derived)) don't change these, they just add reactivity on top, so that when you change a variable something can happen in reaction to it.
+
+As a consequence, the answer to preserving reactivity across boundaries is to use getters/setters or functions (in case of `$state`, `$state.raw` and `$derived`) or an object with mutable properties (in case of `$state`), or a class with reactive properties.
```

Comment:
```suggestion
As a consequence, the answer to preserving reactivity across boundaries is to use getters/setters or functions (in case of `$state`, `$state.raw` and `$derived`), an object with mutable properties (in case of `$state`), or a class with reactive properties.
```